test_app/.gitignore

```diff
@@ -0,0 +1,40 @@
+# See https://help.github.com/articles/ignoring-files for more about ignoring files.
```

Comment:
What is this file about?

Reply:
Sometimes, there are files or folders that you do not want to push to GitHub. For example, node_modules. It is because those files are too many especially on a big project.  By adding a .gitignore file, you can specify what are the files and folders you want to skip while adding them to the staging area and push them to GitHub. 

Some common configurations for the gitignore are listed in https://gist.github.com/octocat/9257657.

Reply:
> It is because those files are too many especially on a big project
So, what you are saying is that if there are too many files, we should ignore some of the files?

Reply:
Thanks for pointing out, please let me stand corrected for that statement. Usually we ignore the files because they are:

- Not used by your project
- Not used by anyone else in your team
- Generated by another process

Some common types of files we would ignore. They are:

- Operating system files
- Application files
- Language and framework files
- Files downloaded with package managers
- Credentials

References: https://zellwk.com/blog/gitignore/